src/datatype/functions_vecf32.rs

```diff
@@ -0,0 +1,290 @@
+#![allow(unused_lifetimes)]
+#![allow(clippy::extra_unused_lifetimes)]
+use crate::datatype::memory_vecf32::{Vecf32Input, Vecf32Output};
+use crate::error::*;
+use base::operator::{Operator, Vecf32Dot};
+use base::scalar::*;
+use base::vector::*;
+use pgrx::pg_sys::Datum;
+use pgrx::pg_sys::Oid;
+use pgrx::pgrx_sql_entity_graph::metadata::ArgumentError;
+use pgrx::pgrx_sql_entity_graph::metadata::Returns;
+use pgrx::pgrx_sql_entity_graph::metadata::ReturnsError;
+use pgrx::pgrx_sql_entity_graph::metadata::SqlMapping;
+use pgrx::pgrx_sql_entity_graph::metadata::SqlTranslatable;
+use pgrx::{FromDatum, IntoDatum};
+use std::alloc::Layout;
+use std::ffi::{CStr, CString};
+use std::ops::{Deref, DerefMut};
+use std::ptr::NonNull;
+
+#[repr(C, align(8))]
+pub struct AccumulateStateHeader {
```

Comment:
It'll be better if it's more generic so that we use reuse it for more usage.

Reply:
I think only `AccumulateState` for `vecf32` and `AccumulateState` for `vecf16` layout can be same?  Just adding a generic type seems won't be so helpful?